Bottom: ca2388b9fec3713cd1c0ee00adb0ce3ac2baf5a0
Top:    c60cf82a11eccdf8eaa2b841748730090969f771
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-25 11:49:27 +0100

Expose draw.c internal drawing APIs


---

diff --git a/src/draw.c b/src/draw.c
index 53757f4..dd10fe0 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -87,10 +87,7 @@ static void DrawPPV (int group, const BoxType *);
 static int DrawLayerGroup (int, const BoxType *);
 static void AddPart (void *);
 static void DrawEMark (ElementTypePtr, LocationType, LocationType, bool);
-static void DrawMask (int side, BoxType *);
-static void DrawPaste (int side, BoxType *);
 static void DrawRats (BoxType *);
-static void DrawSilk (int side, const BoxType *);
 
 
 /*---------------------------------------------------------------------------
@@ -608,7 +605,7 @@ pad_mask_callback (const BoxType * b, void *cl)
  * Draws silk layer.
  */
 
-static void
+void
 DrawSilk (int side, const BoxType * drawn_area)
 {
 #if 0
@@ -667,7 +664,7 @@ DrawMaskBoardArea (int mask_type, BoxType *screen)
 /* ---------------------------------------------------------------------------
  * draws solder mask layer - this will cover nearly everything
  */
-static void
+void
 DrawMask (int side, BoxType * screen)
 {
   int thin = TEST_FLAG(THINDRAWFLAG, PCB) || TEST_FLAG(THINDRAWPOLYFLAG, PCB);
@@ -696,7 +693,7 @@ DrawMask (int side, BoxType * screen)
 /* ---------------------------------------------------------------------------
  * draws solder paste layer for a given side of the board
  */
-static void
+void
 DrawPaste (int side, BoxType *drawn_area)
 {
   gui->set_color (Output.fgGC, PCB->ElementColor);
diff --git a/src/draw.h b/src/draw.h
index cf324a5..5224adc 100644
--- a/src/draw.h
+++ b/src/draw.h
@@ -71,6 +71,10 @@ void EraseElementName (ElementTypePtr);
 void EraseObject (int, void *, void *);
 void LoadBackgroundImage (char *);
 
+void DrawMask (int side, BoxType *);
+void DrawSilk (int side, const BoxType *);
+void DrawPaste (int side, BoxType *);
+
 /*GdkDrawable *draw_get_current_drawable(void);*/
 
 #endif
